book/src/dev/diagrams/mempool-architecture.md

```diff
@@ -0,0 +1,88 @@
+# Mempool Architecture Diagram
+
+This diagram illustrates the architecture of the Zebra mempool, showing its main components and the flow of transactions through the system.
+
+```mermaid
+graph TD
```

Comment:
I am having some issues visualizing the flowchart:

![mem](https://github.com/user-attachments/assets/476125f1-31dd-45be-9546-d036fc8dd5f9)

- Unsupported markdown in some of the descriptions: Remove lists from them.
- Colors in a black background: You can use different node shapes instead of colors: https://mermaid.js.org/syntax/flowchart.html#node-shapes